### PR TITLE
async support for raster data

### PR DESCRIPTION
**Pull request by mmomtchev**
This adds async support for raster data operations

### CODE REVIEW

**Comment by mmomtchev** [comment]
@contra no one is saying anything, so maybe take that bold decision? 😄 

do not hesitate to straighten up my English, especially for the jsdoc

**Comment by mmomtchev** [comment]
#31

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 14.x]
```
Comment (contra) [inline .github/workflows/node.js.yml:19]: Can you add 13 and 15 in here to match what we build against in CI?

**Review by contra — APPROVED** [review]
Overall LGTM - left some minor feedback. Sorry it took a while to review work has been crazy the last few weeks.

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (mmomtchev) [reply .github/workflows/node.js.yml:19]: Sure, GDAL 1.x also needs to be addressed - async is not fully supported on GDAL 1.x but it should at least build

**Comment by mmomtchev** [comment]
Some some reason I had to disable the embedded SQL unit testing on GDAL 1.x, did it work before?

**Comment by contra** [comment]
@mmomtchev Only targeting gdal 2.0+, don't think anything is ever tested or built against 1.x in this project or the original one

**Comment by mmomtchev** [comment]
> @mmomtchev Only targeting gdal 2.0+, don't think anything is ever tested or built against 1.x in this project or the original one

There are AppVeyor platforms configure to build with a shared GDAL 1.x (Ubuntu 16.04 is the one I am testing)

File: test/api_dataset.test.js [hunk 2]
```diff
@@ -530,20 +530,14 @@ describe('gdal.Dataset', () => {
     describe('executeSQL()', () => {
       it('should return Layer', () => {
         const ds = gdal.open(`${__dirname}/data/shp/sample.shp`)
-        if (gdal.version.split('.')[0] < 2) {
-          // TODO: investigate why SQL doesn't work on GDAL 1.x
-          return
-        }
+        console.log(ds)
```
Comment (contra) [inline test/api_dataset.test.js:533]: stray log

**Comment by contra** [comment]
Thanks again - will publish as 2.4.0 once the builds complete.

**Pull request merged**